SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -146,225 +146,286 @@ void ComponentReader::ApplyComponentUpdate(const Worker_ComponentId ComponentId,
 
 	if (UpdatedIds.Num() > 0)
 	{
-		ApplySchemaObject(ComponentObject, Object, Channel, false, UpdatedIds, ComponentId, bOutReferencesChanged);
+		ApplySchemaObject({ ComponentObject, Object, Channel, false, UpdatedIds, ComponentId, bOutReferencesChanged });
 	}
 }
 
-void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject& Object, USpatialActorChannel& Channel, bool bIsInitialData,
-										const TArray<Schema_FieldId>& UpdatedIds, Worker_ComponentId ComponentId,
-										bool& bOutReferencesChanged)
+void ComponentReader::ApplySchemaObject(ApplySchemaObjectDataStruct ApplySchemaObjectData)
 {
-	FObjectReplicator* Replicator = Channel.PreReceiveSpatialUpdate(&Object);
-	if (Replicator == nullptr)
+	ApplySchemaObjectData.Replicator = ApplySchemaObjectData.Channel.PreReceiveSpatialUpdate(&ApplySchemaObjectData.Object);
```

Comment:
I like this design, although I think perhaps it's been done one level too high. Wouldn't it be cleaner to keep the data separate for this function level, but have a ApplySchemaObjectFieldsData that you construct prior to processing the individual fields?

Reply:
Not sure what you mean, do you mean having a second struct?